### DISCUSSION TITLE
Typing in an issue or PR comment box inappropriately scrolls the page on every keystroke

### DISCUSSION THREAD

**Discussion by josh-berry** [community/community#50452] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Bug

### Body

I'm using Firefox 111.0 on macOS.

This seems to only be a thing when writing particularly long comments, but it seems that semi-consistently, when I type a character into a comment box after manually scrolling the page to a particular position, GitHub will scroll the page *back* to whatever its preferred position is, usually with the insertion point just barely visible at the bottom of the window.

Steps to reproduce (sometimes):

1. Start writing a really long comment, and get enough content in the comment box that it has to either expand or show a scroll bar
2. Scroll the page (not the comment box) so that the entire comment box is visible and I can see what I'm writing
3. Type a single character into the comment box
4. Notice that the page scrolls so the entire comment box is no longer visible, and the insertion point is at or near the bottom of the window. I can no longer see anything in the comment box below the insertion point.

This doesn't always seem to happen, but once it starts happening, it happens consistently (on every single keystroke). Switching tabs in the browser (i.e. away from and back to the GitHub page), or switching between "Write" and "Preview" tabs in the comment box, seems to make it stop temporarily. But then it will start happening again randomly for reasons I can't figure out.

Net: This issue makes it really hard to write long, thoughtful comments in GitHub—here's [a recent example](https://github.com/josh-berry/tab-stash/issues/52#issuecomment-1475396447). I had to pull this comment out into VSCode's Markdown editor to finish writing it, because the scrolling behavior was so disruptive it was making me frustrated.

Thanks for looking into this!

**Comment by davenportw** [comment]
replicated issue

**Comment by totten** [comment]
Sometimes, this bug feels like the "Github Uncertainty Principle" - a user may observe the `textarea`, or a user may type in the `textarea`, but the user cannot observe+type at the same time (*lest they provoke insane bounciness*). FWIW, I've seen it with Firefox 111.0 on both MacOS and Ubuntu. 

I haven't replicated the mitigation with switching tabs.

At the moment, the following mitigation is helping me:

1. Add 10-20 blank lines at the bottom of the comment.
2. Scroll the viewport. This is a balancing act:
    * If you can see the textarea's footer-bar ("Attach files by dragging & dropping, selecting or pasting them."), then it will provoke the bounciness.
    * Scroll enough that you push the footer-bar out of the viewport.
    * Scroll enough that you can still see most of the textarea.
3. Put the text-cursor in the place where you want to type and go for it.
4. Keep the blank lines as long as you edit. After you're done editing, delete them.

(It may also help to expand the textarea using the mouse/drag-button - so that you can get a larger visible canvas. Just make sure the footer-bar remains out-of-view while typing.)

At the moment, this seems to be _stable enough_ that you can write some text. It's still annoying if you scroll or cut/paste chunks of prose. But at least you can do some writing.

**Comment by rmartine-ias** [comment]
This has been driving me crazy. Thank you @totten for the workaround. Another I'd like to suggest is [GhostText](https://ghosttext.fregante.com/), which lets you use your text editor instead.
